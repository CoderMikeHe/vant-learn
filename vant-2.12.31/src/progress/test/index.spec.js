import Progress from '..';
import { mount, later } from '../../../test';

test('calc width', async () => {
  const wrapper = mount(Progress, {
    propsData: {
      showPivot: false,
      percentage: 100,
    },
  });
  await later();
  expect(wrapper).toMatchSnapshot();

  wrapper.vm.showPivot = true;
  wrapper.vm.pivotText = 'test';
  await later();
  expect(wrapper).toMatchSnapshot();
});

test('track color prop', async () => {
  const wrapper = mount(Progress, {
    propsData: {
      trackColor: 'green',
    },
  });

  expect(wrapper.element.style.background).toEqual('green');
});
