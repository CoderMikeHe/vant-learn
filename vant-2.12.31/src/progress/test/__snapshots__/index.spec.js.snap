// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calc width 1`] = `<div class="van-progress"><span class="van-progress__portion" style="width: 0px;"></span></div>`;

exports[`calc width 2`] = `<div class="van-progress"><span class="van-progress__portion" style="width: 0px;"><span class="van-progress__pivot" style="left: 0px;">test</span></span></div>`;
